utils/gen-automata/http.py

```diff
@@ -281,6 +282,70 @@ def overwrite_file_header():
     })
 
 
+def auth_value(name):
+    auto = Automaton()
+    start = auto.start()
+    value = auto.add_state()
+
+    start.loop("HorizWhitespace", LabelType.Special)
+    start.loop("\"", LabelType.Char)
+    value.set_name(name)
+    value.mark_enter()
+    line, end = newline()
+    end.loop('HorizWhitespace', LabelType.Special)
+    end.loop(',', LabelType.Char)
+    auto.join(start, line)
+    start.add_transition('All', LabelType.Special, value)
+    value.add_transition('\"', LabelType.Char, end)
+    value.loop_fallback()
+
+    return auto, end, False
+
+
+def authorization_header():
+    """
+    Reads the digest Authorization header
+    Authorization: Digest username="user", realm="Printer API", nonce="dcd98b7102dd2f0e8b11d0f600bfb0c093", uri="/api/version", response="684d849df474f295771de997e7412ea4"
+    """
+
+    # We only really read nonce and response, the rest is assumed for
+    # performance reasons.
+    auth_values = {
+        'nonce': auth_value('nonce'),
+        'response': auth_value('response'),
+    }
+    auto, terminals, add_unknowns = trie(auth_values)
+    start = auto.start()
+    start.set_path("Digest", nocase=True)
```

Comment:
Yes, I think it works kinda by accident, because the unknown handling eats the Digest together with username(the first param) and because we want to ignore username anyway it is OK. This path actually never triggers. Not sure what to do about this though, maybe just put this fact into a comment, so if we in the future want to read `username` whoever implements it will know that this needs to be addressed or somehow make it eat the `Digest` word separately and do it cleanly?